packages/contracts/contracts/universal/MultiSigWallet.sol

```diff
@@ -0,0 +1,513 @@
+        for (uint256 i = 0; i < owners.length; ) {
+            if (confirmation[owners[i]]) {
+                unchecked {
+                    ++count;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+        return count;
+    }
+
+    /**
+     * @inheritdoc IMultiSigWallet
+     */
+    function getTransactionCount(bool _pending, bool _executed) external view returns (uint256) {
+        bool executed;
+        uint256 count = 0;
+        for (uint256 i = 0; i < transactionCount; ) {
+            executed = transactions[i].executed;
+            if ((_pending && !executed) || (_executed && executed)) {
+                unchecked {
+                    ++count;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+        return count;
+    }
+
+    /**
+     * @inheritdoc IMultiSigWallet
+     */
+    function getOwners() external view returns (address[] memory) {
+        return owners;
+    }
+
+    /**
+     * @inheritdoc IMultiSigWallet
+     */
+    function getConfirmations(uint256 _transactionId) external view returns (address[] memory) {
+        address[] memory confirmationsTemp = new address[](owners.length);
+        uint256 count = 0;
+        uint256 i;
+        mapping(address => bool) storage confirmation = confirmations[_transactionId];
+        for (i = 0; i < owners.length; ) {
+            if (confirmation[owners[i]]) {
+                confirmationsTemp[count] = owners[i];
+                unchecked {
+                    ++count;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+        address[] memory _confirmations = new address[](count);
+        for (i = 0; i < count; ) {
+            _confirmations[i] = confirmationsTemp[i];
+
+            unchecked {
+                ++i;
+            }
+        }
+        return _confirmations;
+    }
+
+    /**
+     * @inheritdoc IMultiSigWallet
+     */
+    function getTransactionIds(
+        uint256 _from,
+        uint256 _to,
+        bool _pending,
+        bool _executed
+    ) external view returns (uint256[] memory) {
+        uint256[] memory transactionIdsTemp = new uint256[](transactionCount);
+        uint256 count = 0;
+        bool executed;
+        uint256 i;
+        for (i = 0; i < transactionCount; ) {
+            executed = transactions[i].executed;
+            if ((_pending && !executed) || (_executed && executed)) {
+                transactionIdsTemp[count] = i;
+                unchecked {
+                    ++count;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+        uint256[] memory _transactionIds = new uint256[](_to - _from);
+        for (i = _from; i < _to; ) {
+            _transactionIds[i - _from] = transactionIdsTemp[i];
```

Comment:
If I understand correctly, the length of `transactionIdsTemp` can be different from the length of `transactions`. If `_from` and `_to` mean a range of transaction ids in `transactions`, it seems that this logic can't return the ids of the desired range.

Reply:
The length of transactionIdsTemp is different from transaction. transactionIdsTemp is temporary data for the return value. It serves to filter based on the index between from and to.